src/transformers/configuration_utils.py

```diff
@@ -1026,10 +1026,9 @@ def _get_non_default_generation_parameters(self) -> Dict[str, Any]:
         try:
             default_config = self.__class__()
         except ValueError:
-            for decoder_attribute_name in ("decoder", "generator", "text_config"):
-                if hasattr(self, decoder_attribute_name):
-                    default_config = getattr(self, decoder_attribute_name).__class__()
-                    break
+            decoder_config = self.get_text_config(decoder=True)
+            if decoder_config is not self:
+                default_config = decoder_config.__class__()
```

Comment:
I think the comment above would benefit from an update to clarify what's happening here